rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/parsing/ast/statements.kt

```diff
@@ -1219,6 +1218,46 @@ data class DisplayStmt(val factor1: Expression?, val response: Expression?, over
     }
 }
 
+@Serializable
+data class DOWxxStmt(
+    val comparison: ComparisonOperator,
+    val factor1: Expression,
+    val factor2: Expression,
+    override val body: List<Statement>,
+    override val position: Position? = null
+) : Statement(position), CompositeStatement {
+    fun InterpreterCore.compare(comparison: ComparisonOperator, factor1: Value, factor2: Value, interpreter: InterpreterCore): Boolean {
+        return when (comparison) {
+            ComparisonOperator.EQ -> factor1.compareTo(factor2) == EQUAL
```

Comment:
I had asked you to implement the comparision logic in `operator fun compareTo` of the concrete class because in kotlin this signature allows to compare two instances by using the same operator you can use for scalar type and then, instead to write: `factor1.compareTo(factor2)` you can write `factor1 != factor2` which is much more expressive.
Replace `factor1.compareTo(factor2)` with` factor1 <op> factor2`

Reply:
I found a solution cleaner than actual. While I was analyzing the code for searching the class for this purpose, I saw in `comparison.kt` the `ComparisonOperator?.verify` that takes as input the two factors, interpreter and charset. So, I decided to remove `InterpreterCore.compare` (I believed that was a concrete class) and using  `comparisonOperator.verify` like condition of while in `DOWxxStmt`. Of course, is loses the expressive, but analyzes the cases of `StringValue`, `UnlimitedStringValue` and other, in `comparison.kt` > `compare`. What do you think about this solution? I prepared a new commit.

Reply:
I did not know that function, I agree it is totally unexpressive but it makes its work.